src/lib/Repository/Values/Content/UiRelation.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUi\Repository\Values\Content;
+
+use eZ\Publish\API\Repository\Values\Content\Location;
+use eZ\Publish\Core\Repository\Values\Content\Relation;
+use eZ\Publish\API\Repository\Values\Content\Relation as APIRelation;
+
+/**
+ * Extends original value object in order to provide additional fields.
+ */
+class UiRelation extends Relation
+{
+    /**
+     * Field definition name.
+     *
+     * @var string
+     */
+    protected $fieldDefinitionName;
+
+    /**
+     * The content type name for the destination.
+     *
+     * @var string
+     */
+    protected $destinationContentTypeName;
+
+    /**
+     * The location for the destination.
```

Comment:
The content name is being used for the display. This is used to link to the piece of content: https://github.com/inviqa/hybrid-platform-ui/blob/4438948c96a709a0fa97e5c98adc1f56bde5725c/src/bundle/Resources/views/content/tabs/relations/table.html.twig#L15